google/cloud/internal/populate_common_options_test.cc

```diff
@@ -151,30 +151,52 @@ TEST(PopulateCommonOptions, UniverseDomain) {
       PopulateCommonOptions(Options{}.set<UniverseDomainOption>("my-ud.net"),
                             {}, {}, {}, "default.googleapis.com");
   EXPECT_EQ(actual.get<EndpointOption>(), "default.my-ud.net");
+  EXPECT_EQ(actual.get<AuthorityOption>(), "default.my-ud.net");
 }
 
 TEST(PopulateCommonOptions, EndpointOptionOverridesUniverseDomain) {
   auto actual = PopulateCommonOptions(
       Options{}
-          .set<UniverseDomainOption>("ignored-ud.net")
+          .set<UniverseDomainOption>("test-ud.net")
           .set<EndpointOption>("custom-endpoint.googleapis.com"),
       {}, {}, {}, "default.googleapis.com");
   EXPECT_EQ(actual.get<EndpointOption>(), "custom-endpoint.googleapis.com");
+  EXPECT_EQ(actual.get<AuthorityOption>(), "default.test-ud.net");
+}
+
+TEST(PopulateCommonOptions, AuthorityOptionOverridesUniverseDomain) {
+  auto actual = PopulateCommonOptions(
+      Options{}
+          .set<UniverseDomainOption>("test-ud.net")
+          .set<AuthorityOption>("custom-endpoint.googleapis.com"),
+      {}, {}, {}, "default.googleapis.com");
+  EXPECT_EQ(actual.get<EndpointOption>(), "default.test-ud.net");
+  EXPECT_EQ(actual.get<AuthorityOption>(), "custom-endpoint.googleapis.com");
 }
 
 TEST(PopulateCommonOptions, EnvVarsOverridesUniverseDomain) {
+  ScopedEnvironment authority("AUTHORITY_OPTION",
+                              "authority-env.googleapis.com");
   ScopedEnvironment endpoint("SERVICE_ENDPOINT", "endpoint-env.googleapis.com");
   ScopedEnvironment emulator("SERVICE_EMULATOR", "emulator-env.googleapis.com");
 
   auto actual = PopulateCommonOptions(
-      Options{}.set<UniverseDomainOption>("ignored-ud.net"), "SERVICE_ENDPOINT",
+      Options{}.set<UniverseDomainOption>("test-ud.net"), "SERVICE_ENDPOINT",
       {}, {}, "default.googleapis.com");
   EXPECT_EQ(actual.get<EndpointOption>(), "endpoint-env.googleapis.com");
+  EXPECT_EQ(actual.get<AuthorityOption>(), "default.test-ud.net");
 
   actual = PopulateCommonOptions(
-      Options{}.set<UniverseDomainOption>("ignored-ud.net"), {},
+      Options{}.set<UniverseDomainOption>("test-ud.net"), {},
       "SERVICE_EMULATOR", {}, "default.googleapis.com");
   EXPECT_EQ(actual.get<EndpointOption>(), "emulator-env.googleapis.com");
+  EXPECT_EQ(actual.get<AuthorityOption>(), "default.test-ud.net");
+
+  actual = PopulateCommonOptions(
```

Comment:
optional nit: I might have combined this with the one that passes `"SERVICE_ENDPOINT"` because `"SERVICE_ENDPOINT"` and `"AUTHORITY_ENDPOINT"` are independent.